main_test.go

```diff
@@ -1,154 +1,235 @@
+func TestParseContent_Stdin(t *testing.T) {
+    // Stdin input content
+    content := "Stdin input\n"
+    expected := "Stdin input\n"
+	// Define directText as nil since we are testing stdin only
+	var directText *string
+
+    // Replace stdin with a pipe for testing
+    _, w, cleanup := replaceStdin(t)
+    defer cleanup() // Ensure cleanup is called to restore the original stdin
+
+    // Write content to the pipe and close it
+    go func() {
+        _, err := w.Write([]byte(content))
+        if err != nil {
+            t.Fatalf("Failed to write to stdin pipe: %v", err)
+        }
+        w.Close()
+    }()
+
+    // Execute the function under test with stdin input
+    actual, err := parseContent(directText, []string{})
+    if err != nil {
+        t.Fatalf("TestParseContent_Stdin: Expected no error, got %v", err)
+    }
+
+    // Verify the actual output matches the expected output
+    if actual != expected {
+        t.Errorf("TestParseContent_Stdin: Expected %s but got %s", expected, actual)
+    }
+}
+
+// TestParseContent_File verifies parseContent handles file input correctly.
+func TestParseContent_File(t *testing.T) {
+	// Content to write to the temporary file
+	content := "Content from file"
+	file := createTempFile(t, content)
+
+	// Expected content after parsing the file
+	expected := "Content from file\n"
+
+	// Execute the function under test with the temporary file as input
+	actual, err := parseContent(nil, []string{file.Name()})
+	if err != nil {
+		t.Fatalf("TestParseContent_File: Expected no error, got %v", err)
+	}
+
+	// Verify the actual output matches the expected output
+	if actual != expected {
+		t.Errorf("TestParseContent_File: Expected %s but got %s", expected, actual)
+	}
+}
+
+// stringPtr creates a pointer to a string.
+func stringPtr(s string) *string {
+	return &s
+}
+
+// createTempFile creates a temporary file for testing purposes and writes the given content to it.
+func createTempFile(t *testing.T, content string) *os.File {
+t.Helper ()
+	// Create a temporary file
+	file, err := os.CreateTemp(t.TempDir(), "testfile")
+	if err != nil {
+		t.Fatalf("createTempFile: Failed to create temp file: %v", err)
+	}
+
+	// Write the provided content to the temporary file
+	_, err = file.WriteString(content)
+	if err != nil {
+		t.Fatalf("createTempFile: Failed to write to temp file: %v", err)
+	}
+
+	return file
+}
+
+// replaceStdin replaces os.Stdin with a pipe for testing purposes.
+func replaceStdin(t *testing.T) (*os.File, *os.File, func()) {
+t.Helper ()
+    // Create a pipe
+    r, w, err := os.Pipe()
+    if err != nil {
+        t.Fatalf("replaceStdin: Failed to create pipe: %v", err)
+    }
+
+    // Save the original stdin
+    originalStdin := os.Stdin
+
+    // Replace stdin with the read end of the pipe
+    os.Stdin = r
+
+    // Cleanup function to restore the original stdin and close the pipe
+    cleanup := func() {

```

Comment:
Did you test this?

This is what I suggested, I assumed it would work

```suggestion
    t.Cleanup (func() {
```

Reply:
this would allow to remove the  `defer cleanup() ` when using the function

Reply:
Fixed by #21